src/Activation/ActivateProximityTracing.tsx

```diff
@@ -24,7 +24,8 @@ const ActivateProximityTracing: FunctionComponent = () => {
   const navigation = useNavigation()
   const { applicationName } = useApplicationName()
   const { exposureNotifications } = usePermissionsContext()
-  const { isLocationOffAndNeeded } = useHasLocationRequirements()
+  const { isLocationOn, isLocationNeeded } = useActivationContext()
+  const isLocationOffAndNeeded = !isLocationOn && isLocationNeeded
```

Comment:
I had it that way at first, but it introduced the possibility of impossible shenanigans like…

```
{
  isLocationOn: false,
  isLocationNeeded: true,
  isLocationOffAndNeeded: false,
}

… when mocking the provider. Is there a way around that?

Reply:
Sadly, I don't think there is unless we change the `isLocationOffAndNeeded` from a property to a function that returns:

``` typescript
const isLocationOffAndNeeded = (): boolean => {
 return !isLocationOn && isLocationNeeded;
}
```
Even though the nature of how context providers are defined you could still override the function when creating the provider and this will make uses of that property a bit weird since we would have to do `isLocationOffAndNeeded()` to use it, probably not worth it.

Reply:
Thanks for the explanation. I wish there were a better way to handle stuff like this, as we've run into this exact dilemma on a few occasions now.